File: data_structures/stack/go/stack.go
package main

type stack struct {
	data []interface{}
}

func (s *stack) Push(value interface{}) {
	s.data = append(s.data, value)
}

func (s *stack) Pop() interface{} {
	//get last item
	value := s.data[len(s.data)-1]
	//shrink array
	s.data = s.data[0 : len(s.data)-1]

	return value
}

File: data_structures/stack/php/stack.php
<?php 
class Stack { 
    private $_data = array();
    private $_top = -1; 
    public function Pop(){
        if($this->_top == -1){
            throw new Exception("Stack is empty");
        }else{
            $this->_top--;
        }
    }
    public function Push($data) { 

        $this->_top++;
        array_push($this->_data,$data);
    } 
    public function Top(){
        return $this->_data[$this->_top];
    }
    
} 

$stack = new Stack(); 
$stack->Push("First Input");
$stack->Push("Second Input");

echo $stack->Top();
$stack->Pop();
echo $stack->Top();
?> 
